server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java

```diff
@@ -260,39 +222,40 @@ private PhysicalPlan transformQuery(QueryOperator queryOperator)
               // get pathForDataType and measurementColumn
               String measurement = path.getMeasurement();
               String pathForDataType;
-              String measurementColumn;
+              String measurementChecked;
+              // check datatype consistency
+              String originAggregation = null;
               if (originAggregations != null && !originAggregations.isEmpty()) {
                 pathForDataType = originAggregations.get(i) + "(" + path.getFullPath() + ")";
-                measurementColumn = originAggregations.get(i) + "(" + measurement + ")";
+                measurementChecked = originAggregations.get(i) + "(" + measurement + ")";
               } else {
                 pathForDataType = path.getFullPath();
-                measurementColumn = measurement;
+                measurementChecked = measurement;
```

Comment:
what is measurementChecked?   Rename to measurementwithaggregation is better

Reply:
MeasurementChecked means 'the measurements during the datatype check'. Keep it.